src/plugins/MaxCheck.cpp

```diff
@@ -90,6 +90,16 @@ void MaxCheck::readXML(XMLfileUnits& xmlconfig) {
 
 }
 
+void MaxCheck::siteWiseForces(
+		ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
+		unsigned long simstep)
+{
+	if (simstep < _control.start || simstep > _control.stop
+			|| simstep % _control.freq != 0)
+		return;
+	this->checkMaxVals(particleContainer, domainDecomp, simstep);
```

Comment:
should this be deleted for endStep and afterForces, or do you want to have that everywhere?

Reply:
Currently, only the resulting force on a molecule is considered. This works for mono-atomic ones, but for molecular models with multiple sites we should limit the site-wise forces to avoid too high rotation speeds.

Since the velocity is calculated at two code locations, we need also two calls to the function that ensures the speed limit. So, one call we already have with: plugin->siteWiseForces(), the best location for the second call I guess would be here: plugin->beforeEventNewTimestep(), so right before new velocity and new particle position is being calculated.

 I will change and test that accordingly.